### PR TITLE
Mark `ServerRequestFactory` as deprecated + Refactoring

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #52

### CODE REVIEW

File: src/Exception/BadRequestException.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Runner\Http\Exception;
+
+use Exception;
+
+final class BadRequestException extends Exception
```
Comment (samdark) [inline src/Exception/BadRequestException.php:9]: How would emitter react on throwing this exception? Would it properly respond with HTTP code 400?

**Comment by samdark** [comment]
The parsing itself and overall logic is 👍

File: src/Exception/BadRequestException.php (hunk 1 above)
Comment (vjik) [reply src/Exception/BadRequestException.php:12]: Done. I had to remake everything :)

**Review by samdark — APPROVED** [review]

File: src/HttpApplicationRunner.php [hunk 2]
```diff
@@ -5,6 +5,7 @@
 namespace Yiisoft\Yii\Runner\Http;
 
 use ErrorException;
+use HttpSoft\Message\ResponseFactory;
```
Comment (xepozz) [inline src/HttpApplicationRunner.php:8]: I don't think that the package should be linked with that package directly

**Review by xepozz — CHANGES_REQUESTED** [review]

File: src/HttpApplicationRunner.php (hunk 2 above)
Comment (vjik) [reply src/HttpApplicationRunner.php]: Done

**Comment by samdark** [comment]
The changes in public API:

- Removed `public function createFromGlobals(): ServerRequestInterface`.
- Removed `public function createFromParameters(array $server, array $headers = [], array $cookies = [], array $get = [], array $post = [], array $files = [], mixed $body = null): ServerRequestInterface`.
- Added `public function create($body = null): ServerRequestInterface`.

**Pull request merged**